File: src/main/java/eu/qwsome/sql/condition/BiCondition.java
package eu.qwsome.sql.condition;


/**
 * @author Lukáš Kvídera
 */
abstract class BiCondition implements Condition {

  /**
   * The colum on the left side of operator.
   */
  private final ValueHolder first;

  /**
   * The colum on the right side of operator.
   */
  private final ValueHolder second;

  BiCondition(final ValueHolder first, final ValueHolder second) {
    this.first = first;
    this.second = second;
  }

  @Override
  public ValueConstructor getValues() {
    return new ValueConstructor()
      .add(this.first)
      .add(this.second);
  }

  @Override
  public void appendTo(final StringBuilder builder) {
    builder.append(this.first.getSql())
      .append(getOperator())
      .append(this.second.getSql());
  }

  abstract String getOperator();
}

File: src/main/kotlin/eu/qwsome/sql/condition/BiCondition.kt
/*
 * @author Lukáš Kvídera
 */
package eu.qwsome.sql.condition

abstract class BiCondition(
    /**
     * The column on the left side of operator.
     */
    private val first: ValueHolder,
    /**
     * The column on the right side of operator.
     */
    private val second: ValueHolder,
) : Condition {

    override fun getValues(): ValueConstructor = ValueConstructor().addAll(first, second)

    override fun appendTo(builder: StringBuilder) {
        builder.append(first.getSql())
            .append(getOperator())
            .append(second.getSql())
    }

    protected abstract fun getOperator(): String
}
